homeassistant/components/mqtt/models.py

```diff
@@ -51,6 +51,22 @@ class PayloadSentinel(StrEnum):
 type PublishPayloadType = str | bytes | int | float | None
 
 
+def convert_outgoing_mqtt_payload(
+    payload: PublishPayloadType,
+) -> PublishPayloadType:
+    """Ensure correct raw MQTT payload is passed as bytes for publishing."""
+    if isinstance(payload, str):
```

Comment:
Should we also check the string starts with `b'` or `b"`, since that's all we care about?
```suggestion
    if isinstance(payload, str) and payload.startswith(("b'", 'b"')):
```